src/accelerate/utils/environment.py

```diff
@@ -121,3 +129,41 @@ def check_fp8_capability():
     """
     cuda_device_capacity = torch.cuda.get_device_capability()
     return cuda_device_capacity >= (8, 9)
+
+
+def get_query_command(device_id: int = None, backend: str = None):
+    """
+    Returns the right param args for `numa` support based on a `backend`.
+
+    Currently only supports `nvidia`.
+    """
+    if backend == "nvidia":
+        return [_get_nvidia_smi(), "--query-gpu=pci.bus_id", "-i", str(device_id), "--format=csv,noheader"]
+    else:
+        # TODO: Add support for AMD and Gaudi2
+        raise ValueError(f"Backend {backend} is currently not supported through `accelerate launch`.")
+
+
+def get_bus_id(device_id: int = None, backend: str = None):
+    """
+    Gets the bus ID for `device_id`. If not passed, will default to the local rank.
+    """
+    if device_id is None:
+        device_id = os.environ.get("LOCAL_RANK", 0)
+    command = get_query_command(device_id, backend)
+    output = subprocess.check_output(command, universal_newlines=True)
+    # The output is a string like "00000000:00:04.0"
+    # We want to return "0000:00:04.0"
+    # As it can be passed to `ls -l /sys/bus/pci/devices/${bus_id}`
+    return output.strip()[4:]
+
+
+def get_numa_node_for_device(device_id: int = None, backend: str = None):
+    """
+    Gets the numa node for `device_id`. If not passed, will default to the local rank.
+    """
+    bus_id = get_bus_id(device_id, backend)
+    if bus_id is None:
```

Comment:
`get_bus_id` will always return a string unless it errors, right? So this is dead code.